src/lightning_app/core/app.py

```diff
@@ -79,6 +82,10 @@ def __init__(
 
         _validate_root_flow(root)
         self._root = root
+        self.default_cloud_compute = default_cloud_compute or lightning_app.CloudCompute()
+
+        if self.default_cloud_compute.preemptible:
+            raise Exception("The default Cloud Compute can't be preemptible.")
```

Comment:
Note: it is being removed here:
https://github.com/Lightning-AI/lightning/pull/14852/